frappe/core/doctype/communication/communication.py

```diff
@@ -351,16 +351,26 @@ def get_contacts(email_strings):
 		email = get_email_without_link(email)
 		contact_name = get_contact_name(email)
 
-		if not contact_name:
-			contact = frappe.get_doc({
-				"doctype": "Contact",
-				"first_name": frappe.unscrub(email.split("@")[0]),
-			})
-			contact.add_email(email_id=email, is_primary=True)
-			contact.insert(ignore_permissions=True)
-			contact_name = contact.name
-
-		contacts.append(contact_name)
+		if not contact_name and email:
+			email_parts = email.split("@")
+			first_name = frappe.unscrub(email_parts[0])
+
+			try:
+				contact = frappe.get_doc({
+					"doctype": "Contact",
+					"first_name": first_name,
+				})
+				contact.add_email(email_id=email, is_primary=True)
+				contact.name = ('{0}-{1}'.format(first_name, email_parts[1])
+					if first_name == 'Contact' else first_name)
```

Comment:
since `contact` is not equal to `Contact`
```suggestion
					if first_name.lower() == 'contact' else first_name)
```

Reply:
frappe.unscrub("contact') returns "Contact"